src/Sarif.Sarifer/GenerateTestDataCommand.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel.Design;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Sarif.Viewer.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    internal class GenerateTestDataCommand : MenuCommand
+    {
+        private static SarifViewerInterop s_viewerInterop;
+
+        public GenerateTestDataCommand(IVsShell vsShell) :
+            base(
+                new EventHandler(MenuCommandCallback),
+                new CommandID(Guids.SariferCommandSet, SariferPackageCommandIds.GenerateTestData))
+        {
+            s_viewerInterop = new SarifViewerInterop(vsShell);
+        }
+
+        /// <summary>
+        /// Event handler called when the user selects the Generate SARIF Test Data command.
+        /// </summary>
+        /// <remarks>
+        /// Since this is a menu item callback, it must return void.
+        /// </remarks>
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private static async void MenuCommandCallback(object caller, EventArgs args)
```

Comment:
>static async [](start = 16, length = 12)

This doesn't need to be static or async :)

Use VS's FileAndForget method for calling OpenSarifLogsasync().FileAndForget() for starting the task.

Reply:
Fixed. The reason it wasn't compiling was I needed a reference to separate NuGet package `Microsoft.VisualStudio.Shell.15.0`.

---
In reply to: [514614472](https://github.com/microsoft/sarif-visualstudio-extension/pull/234#discussion_r514614472) [](ancestors = 514614472)